src/biogeochem/CNCStateUpdate1Mod.F90

```diff
@@ -60,23 +64,34 @@ subroutine CStateUpdateDynPatch(bounds, num_soilc_with_inactive, filter_soilc_wi
     associate( &
          cf_veg => cnveg_carbonflux_inst  , &
          cs_veg => cnveg_carbonstate_inst , &
+         cf_soil => soilbiogeochem_carbonflux_inst, &
          cs_soil => soilbiogeochem_carbonstate_inst &
          )
 
     dt = get_step_size_real()
-
     if (.not. use_fates) then
        do j = 1,nlevdecomp
           do fc = 1, num_soilc_with_inactive
              c = filter_soilc_with_inactive(fc)
-             cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) + &
+!             if (.not. use_soil_matrixcn) then
+                cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) + &
                   cf_veg%dwt_frootc_to_litr_met_c_col(c,j) * dt
-             cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) + &
+                cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) + &
                   cf_veg%dwt_frootc_to_litr_cel_c_col(c,j) * dt
-             cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) + &
+                cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) + &
                   cf_veg%dwt_frootc_to_litr_lig_c_col(c,j) * dt
-             cs_soil%decomp_cpools_vr_col(c,j,i_cwd) = cs_soil%decomp_cpools_vr_col(c,j,i_cwd) + &
+                cs_soil%decomp_cpools_vr_col(c,j,i_cwd) = cs_soil%decomp_cpools_vr_col(c,j,i_cwd) + &
                   ( cf_veg%dwt_livecrootc_to_cwdc_col(c,j) + cf_veg%dwt_deadcrootc_to_cwdc_col(c,j) ) * dt
+!             else
+!                cf_soil%matrix_input_col(c,j,i_met_lit) = cf_soil%matrix_input_col(c,j,i_met_lit) + &
+!                     cf_veg%dwt_frootc_to_litr_met_c_col(c,j) * dt
+!                cf_soil%matrix_input_col(c,j,i_cel_lit) = cf_soil%matrix_input_col(c,j,i_cel_lit) + &
+!                     cf_veg%dwt_frootc_to_litr_cel_c_col(c,j) * dt
+!                cf_soil%matrix_input_col(c,j,i_lig_lit) = cf_soil%matrix_input_col(c,j,i_lig_lit) + &
+!                     cf_veg%dwt_frootc_to_litr_lig_c_col(c,j) * dt
+!                cf_soil%matrix_input_col(c,j,i_cwd)     = cf_soil%matrix_input_col(c,j,i_cwd) + &
+!                    (cf_veg%dwt_livecrootc_to_cwdc_col(c,j) + cf_veg%dwt_deadcrootc_to_cwdc_col(c,j) ) * dt
+!             end if !soil_matrix
```

Comment:
Please remove all the commented out lines. also, there is another commented out line in line 76.

Reply:
done